test/integration/scoped_rds_integration_test.cc

```diff
@@ -0,0 +1,219 @@
+#include "envoy/api/v2/srds.pb.h"
+
+#include "common/config/resources.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/http_integration.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class ScopedRdsIntegrationTest : public HttpIntegrationTest,
+                                 public Grpc::GrpcClientIntegrationParamTest {
+protected:
+  struct FakeUpstreamInfo {
+    FakeHttpConnectionPtr connection_;
+    FakeUpstream* upstream_{};
+    FakeStreamPtr stream_;
+  };
+
+  ScopedRdsIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, ipVersion(), realTime()) {}
+
+  ~ScopedRdsIntegrationTest() override {
+    resetConnections();
+    cleanupUpstreamAndDownstream();
+  }
+
+  void initialize() override {
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+      // Add the static cluster to serve SRDS.
+      auto* scoped_rds_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      scoped_rds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      scoped_rds_cluster->set_name("srds_cluster");
+      scoped_rds_cluster->mutable_http2_protocol_options();
+
+      // Add the static cluster to serve RDS.
+      auto* rds_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      rds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      rds_cluster->set_name("rds_cluster");
+      rds_cluster->mutable_http2_protocol_options();
+    });
+
+    config_helper_.addConfigModifier(
+        [this](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager&
+                   http_connection_manager) {
+          const std::string& scope_key_builder_config_yaml = R"EOF(
+fragments:
+  - header_value_extractor: { name: X-Google-VIP }
+)EOF";
+          envoy::config::filter::network::http_connection_manager::v2::ScopedRoutes::ScopeKeyBuilder
+              scope_key_builder;
+          MessageUtil::loadFromYaml(scope_key_builder_config_yaml, scope_key_builder);
```

Comment:
Can you merge master to pick up changes? This will need to change to `TestUtil::loadFromYaml`..

Reply:
Done.